src/server/cea_608_encoder/cea_608_utility.py

```diff
@@ -56,19 +56,74 @@ def bytes_to_byte_pairs(byte_list: list) -> list:
     return byte_pairs
 
 
-def create_byte_pairs(caption_string: str) -> list:
+def which_char_set(caption_char: str) -> str:
+    """Finds which character set the letter is in
+
+    :param caption_char:
+    :return: the name of the char set the caption_char is in
+    """
+    if caption_char in char_sets['basic_na_set']:
+        return 'basic_na_set'
+    elif caption_char in char_sets['special_na_set']:
+        return 'special_na_set'
+    elif caption_char in char_sets['extended_we_sm_set']:
+        return 'extended_we_sm_set'
+    elif caption_char in char_sets['extended_we_french_set']:
+        return 'extended_we_french_set'
+    elif caption_char in char_sets['extended_we_port_set']:
+        return 'extended_we_port_set'
+    elif caption_char in char_sets['extended_we_gd_set']:
+        return 'extended_we_gd_set'
+
+
+def which_channel(channel_toggle: int,char_set: str) -> str:
+    """Provides the correct first byte to a letter depending on the channel toggle
+
+    :param channel_toggle, char_set:
+    :return: The first byte
+    """
+    if channel_toggle == 0:
+        if char_set == 'special_na_set':
+            return '11'
+        elif char_set == 'extended_we_sm_set' or char_set == 'extended_we_french_set':
+            return '12'
+        elif char_set == 'extended_we_port_set' or char_set == 'extended_we_gd_set':
+            return '13'
+        else:
```

Comment:
Instead of returning `-1` to signify the char set wasn't located, what's your opinion on if the char set isn't found we raise an error to tell the user that they gave us a character for which we can't create a byte pair for? Maybe something like `raise ValueError(f'The character: {char_set} is not in any of the supported character sets')`

If you can think of something else please share :)

Reply:
That sounds good! Adding it now.

Reply:
When we call which_char_set() there is no final else statement so we could possibly get nothing back from that function